modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1515,17 +1515,31 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
   /**
    * Get the current market price from a third party to be used for recovery
    */
-  getRecoveryMarketPrice(): Bluebird<string> {
+   getRecoveryMarketPrice(): Bluebird<string> {
     const self = this;
     return co<string>(function *getRecoveryMarketPrice() {
-      const bitcoinAverageUrl = config.bitcoinAverageBaseUrl + self.getFamily().toUpperCase() + 'USD';
-      const response = yield request.get(bitcoinAverageUrl).retry(2).result();
-
-      if (response === null || typeof response.last !== 'number') {
-        throw new Error('unable to reach BitcoinAverage for price data');
+      // TODO This map should be moved to the place where the list of families is defined
+      const familyNamesToCoinGeckoIds = new Map()
+        .set('BTC', 'bitcoin')
+        .set('LTC', 'litecoin')
+        .set('BCH', 'bitcoin-cash')
+        .set('ZEC', 'zcash')
+        .set('DASH', 'dash');
+      const coinGeckoId = familyNamesToCoinGeckoIds.get(self.getFamily().toUpperCase());
+      const coinGeckoUrl = config.coinGeckoBaseUrl + `simple/price?ids=${coinGeckoId}&vs_currencies=USD`;
+      const response = yield request.get(coinGeckoUrl).retry(2).result();
+
+      // An example of response
+      // {
+      //   "ethereum": {
+      //     "usd": 220.64
+      //   }
+      // }
+      if (!response || typeof response[coinGeckoId]['usd'] !== 'number') {
```

Comment:
Should both these cases be treated the same?

Reply:
There are 2 separate checks now, originally I tried to keep logic the same as was written before.